make test fails when jetstream server is not running

**Describe the bug**

Jetstream required for unit tests but not documented as required tool.

**To Reproduce**
Steps to reproduce the behavior:

1. Install the required tools - https://github.com/numaproj/numaflow/blob/main/docs/development.md#install-required-tools
2. Run `make test`
3. Failed test with the following output:
```
    processor_manager_test.go:35: 
        	Error Trace:	processor_manager_test.go:35
        	Error:      	Received unexpected error:
        	            	failed to connect to nats url=nats://127.0.0.1:4222: nats: no servers available for connection
        	Test:       	TestFetcherWithSameOTBucket
--- FAIL: TestFetcherWithSameOTBucket (0.00s)
```

**Expected behavior**
`make test` to pass

**Environment (please complete the following information):**

- Kubernetes: N/A
- Numaflow: head of main

---

<!-- Issue Author: Don't delete this message to encourage other users to support your issue! -->

**Message from the maintainers**:

Impacted by this bug? Give it a 👍. We often sort issues this way to know what to prioritize.

--- a/pkg/watermark/fetch/processor_manager_test.go
+++ b/pkg/watermark/fetch/processor_manager_test.go
@@ -1,4 +1,4 @@
-// //go:build isb_jetstream
+//go:build isb_jetstream
 
 package fetch
 

